app/recurring_job.go

```diff
@@ -111,6 +113,13 @@ func recurringJob(c *cli.Context) (err error) {
 	var jobConcurrent int = recurringJob.Spec.Concurrency
 	jobTask := recurringJob.Spec.Task
 
+	recurringJob.Status.ExecutionCount += 1
+	if _, err = lhClient.LonghornV1beta2().RecurringJobs(namespace).UpdateStatus(context.TODO(), recurringJob, metav1.UpdateOptions{}); err != nil {
+		return errors.Wrap(err, "failed to update job execution count")
+	}
+
+	jobExecutionCount := recurringJob.Status.ExecutionCount
```

Comment:
It is a strictly increasing integer, right?

Reply:
Yes, this `ExecutionCount` is only updated here
and it will first update the status immediately after +1

```
	recurringJob.Status.ExecutionCount += 1
	if _, err = lhClient.LonghornV1beta2().RecurringJobs(namespace).UpdateStatus(context.TODO(), recurringJob, metav1.UpdateOptions{}); err != nil {
		return errors.Wrap(err, "failed to update job execution count")
	}
        jobExecutionCount := recurringJob.Status.ExecutionCount
```